File: Web/NthFormatter.cs
using System.Globalization;

namespace Web
{
    public class NthFormatter
    {
        private readonly string _n;

        public NthFormatter(int n)
        {
            _n = n.ToString(CultureInfo.InvariantCulture);
        }

        public string Format()
        {
            if (FormatsAsFirst)
                return $"{_n}st";
            if (FormatsAsSecond)
                return $"{_n}nd";
            if (FormatsAsThird)
                return $"{_n}rd";
            return $"{_n}th";
        }

        private bool FormatsAsFirst => _n.EndsWith("1") && !_n.EndsWith("11");
        private bool FormatsAsSecond => _n.EndsWith("2") && !_n.EndsWith("12");
        private bool FormatsAsThird => _n.EndsWith("3") && !_n.EndsWith("13");
    }
}

File: Web/app/scripts/nth-formatter.ts
function shouldFormatsAsFirst(s: string) {
    return s.endsWith('1') && !s.endsWith('11');
}

function shouldFormatsAsSecond(s: string) {
    return s.endsWith('2') && !s.endsWith('12');
}

function shouldFormatsAsThird(s: string) {
    return s.endsWith('3') && !s.endsWith('13');
}

export default {
    format(n: number) {
        const s = n.toString();
        if (shouldFormatsAsFirst(s))
            return `${s}st`;
        if (shouldFormatsAsSecond(s))
            return `${s}nd`;
        if (shouldFormatsAsThird(s))
            return `${s}rd`;
        return `${s}th`;
    }
};
